docs/kola/external-tests.md

```diff
@@ -264,6 +264,12 @@ is simple and is not expected to conflict with other tests, it should be marked
 `exclusive: false`. When the `exclusive` key is not provided, tests are marked
 `exclusive: true` by default.
 
+The `isolation` key can take two values: `readonly` and `dynamicuser`.  These
+are thin wrappers for the equivalent systemd options; `readonly` equals `ProtectSystem=strict`,
+and `dynamicuser` means `DynamicUser=yes`.  Setting either of these options
+also implies `exclusive: false`.  Use these for tests that are mostly about
```

Comment:
I really think it's best if we don't rely on `exclusive: false` being `implicit` anywhere. Tests can do things in their butane config that are exclusive and would/could cause other tests to fail.

If you want an `isolation` and `dynamicuser` flag that's OK I guess, but I really want to keep `exlusive: false` explicit.

Reply:
Hmm, we could also enforce that such tests don't provide an ignition config.

Reply:
I'm not really opposed to also requiring an explicit `exclusive: false` as long as we don't force people to copy-paste the rationale for it :smile:

Reply:
Now strengthened this to prevent the test from providing an Ignition config too!

Reply:
I did also test the testing framework manually:

```
$ cat > src/config/tests/kola/content-origins/config.bu << 'EOF'
variant: fcos
version: 1.2.0
EOF
$ kola run ext.config.content-origins
Error: test ext.config.content-origins specifies isolation=dynamicuser but includes an Ignition config
2022-09-01T14:09:32Z cli: test ext.config.content-origins specifies isolation=dynamicuser but includes an Ignition config
$ 
```

Reply:
> I'm not really opposed to also requiring an explicit `exclusive: false` as long as we don't force people to copy-paste the rationale for it smile

That's fine. It's really not that hard to include a rationale, though, is it? Also there can always be more context provided in these comments, it doesn't just have to be a copy and paste. For example, for tests that do have a butane config they could explain why the butane config changes are safe for `exclusive: false`.

Reply:
Why would I add a comment/rationale for that?  Would you block https://github.com/coreos/fedora-coreos-config/pull/1949 on that?  If so I'm sorry I just disagree, it's self-evidently obvious and doesn't need a comment.

Reply:
I think the problem is that a lot of the configuration options here aren't self-evident especially to someone who didn't write the test and it's safer to just default to including a rationale for everything to make sure we catch the full context.

I agree with you that if you read the documentation on this one, and then you read the systemd documentation on that option, then it's clear: "this test is completely read-only and doesn't write to the filesystem at all", but that requires... work, we could just state it up front. It doesn't really cost us anything to help the user read/understand what's going on.

Reply:
> Would you block [coreos/fedora-coreos-config#1949](https://github.com/coreos/fedora-coreos-config/pull/1949) on that?

And.. the answer is "no", I'm not that dogmatic. I'd suggest it but if you firmly say "are you going to block this on that" I wouldn't, but if I ever touched the file I'd be like "hmm, wonder why these options aren't documented like all the others?" and I'd add it back.

Reply:
OK, I think we can continue a debate about the comment schema after landing this.